py/web/views/events.py

```diff
@@ -901,22 +901,32 @@ async def execute(self, m: Model):
         )
 
 
+class GroupUpdateMessage(BaseModel):
+    ticketType: int
+    message: str
+
+
 class EventUpdate(UpdateView):
     class Model(BaseModel):
         subject: constr(max_length=200)
-        message: str
+        message: Optional[str]
+        groupMessages: Optional[List[GroupUpdateMessage]]
 
     async def check_permissions(self):
         await check_session(self.request, 'admin', 'host')
 
     async def execute(self, m: Model):
         event_id = await _check_event_permissions(self.request, check_upcoming=True)
+
+        if not (m.message or m.groupMessages):
+            raise JsonErrors.HTTPBadRequest(message='You need to provide at least one message')
```

Comment:
you can move this into a validator, see [here](https://pydantic-docs.helpmanual.io/usage/validators/)

Reply:
ty